### PR TITLE
apache-archiva: fix non-native binaries 

### PR DESCRIPTION
**Pull request by iMichka**
This just ships a bunch of pre-compiled binaries and is not built from source.
This was already known in https://github.com/Homebrew/homebrew-core/pull/12427#issuecomment-294190196

Fixes:
    * Binaries built for a non-native architecture were installed into apache-archiva's prefix.
      The offending files are:
        /opt/homebrew/Cellar/apache-archiva/2.2.5_1/libexec/bin/wrapper-macosx-universal-32	(universal)
        /opt/homebrew/Cellar/apache-archiva/2.2.5_1/libexec/bin/wrapper-macosx-universal-64	(universal)
        /opt/homebrew/Cellar/apache-archiva/2.2.5_1/libexec/lib/libwrapper-macosx-universal-32.jnilib	(universal)
        /opt/homebrew/Cellar/apache-archiva/2.2.5_1/libexec/lib/libwrapper-macosx-universal-64.jnilib	(universal)

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
We can probably build those wrappers from source. See #88380.

**Comment by iMichka** [comment]
Note for myself:
Commercial website / license pages: https://wrapper.tanukisoftware.com/doc/english/licenseOverview.html
Source code can be found here: https://sourceforge.net/p/wrapper/code/2064/tree/trunk/wrapper/

I'll have a look if I can build this from source ...

**Comment by iMichka** [comment]
The "drawback" is as this needs openjdk 8, we sort of "loose" the ARM build. On the other hand this was an `all` bottle, so who knows if it was working properly.

**Comment by danielnachun** [comment]
It's probably worth noting here that Azul now ships an OpenJDK8 ARM build: https://cdn.azul.com/zulu/bin/zulu8.58.0.13-ca-jdk8.0.312-macosx_aarch64.tar.gz.  I also opened a PR to try to fix OpenJDK8 on Intel Monterey: https://github.com/Homebrew/homebrew-core/pull/89982.  I'll try to force an ARM build on this if it works with Intel.

File: Formula/apache-archiva.rb [hunk 1]
```diff
@@ -4,17 +4,42 @@ class ApacheArchiva < Formula
...
     (bin/"archiva").write_env_script libexec/"bin/archiva", JAVA_HOME: Formula["openjdk"].opt_prefix
+
+    resource("wrapper").stage do
+      # Use openjdk@8, fixes: error: Target option 1.4 is no longer supported. Use 7 or later.
+      ENV["JAVA_HOME"] = Formula["openjdk@8"].opt_prefix
+      ENV["ANT_HOME"] = Formula["ant"].opt_libexec
+      system "./build64.sh"
+      libexec.install "bin/wrapper" => "#{libexec}/bin/wrapper"
+      libexec.install "lib/libwrapper.jnilib" => "#{libexec}/lib/libwrapper.jnilib" if OS.mac?
+      libexec.install "lib/libwrapper.so" => "#{libexec}/lib/libwrapper.so" if OS.linux?
```
Comment (carlocab) [inline Formula/apache-archiva.rb:40]:
```suggestion
      libext = OS.mac? ? "jnilib" : "so"
      (libexec/"lib").install "lib/libwrapper.#{libext}"
```

**Comment by iMichka** [comment]
I also added `depends_on arch: :x86_64 # openjdk@8 doesn't support ARM` as for other formulae.

**Comment by carlocab** [comment]
Maybe this can wait until #90008 is merged.

File: Formula/apache-archiva.rb [hunk 2]
```diff
@@ -4,17 +4,28 @@ class ApacheArchiva < Formula
...
-  license "Apache-2.0"
+  license all_of: ["Apache-2.0", "GPL-2.0-only"]
+  revision 1
 
   bottle do
     sha256 cellar: :any_skip_relocation, all: "7be56a181bec0957d8f4f2936e99a98f737b3f7e146a3f7384b13d9e824919ad"
   end
 
+  depends_on "ant" => :build
+  depends_on "java-service-wrapper" => :build
```
Comment (carlocab) [inline Formula/apache-archiva.rb:15]: Can we use symlinks instead of copying stuff over, and make this a runtime dependency?

Comment (iMichka) [reply Formula/apache-archiva.rb]: Done

File: Formula/apache-archiva.rb [hunk 3]
```diff
@@ -4,17 +4,28 @@ class ApacheArchiva < Formula
...
   end
 
+  depends_on "ant" => :build
+  depends_on "java-service-wrapper"
   depends_on "openjdk"
 
   def install
     libexec.install Dir["*"]
+    rm_f Dir["#{libexec}/bin/wrapper*"]
+    rm_f Dir["#{libexec}/lib/libwrapper*"]
```
Comment (carlocab) [inline Formula/apache-archiva.rb:21]:
```suggestion
    rm_f libexec.glob("bin/wrapper*")
    rm_f libexec.glob("lib/libwrapper*")
```

File: Formula/apache-archiva.rb [hunk 4]
```diff
@@ -4,17 +4,28 @@ class ApacheArchiva < Formula
...
     (bin/"archiva").write_env_script libexec/"bin/archiva", JAVA_HOME: Formula["openjdk"].opt_prefix
+
+    wrapper = Formula["java-service-wrapper"].opt_libexec
+    ln_s wrapper/"bin/wrapper", "#{libexec}/bin/wrapper"
+    libext = OS.mac? ? "jnilib" : "so"
+    ln_s wrapper/"lib/libwrapper.#{libext}", "#{libexec}/lib/libwrapper.#{libext}"
+    ln_s wrapper/"lib/wrapper.jar", "#{libexec}/lib/wrapper.jar"
```
Comment (SMillerDev) [inline Formula/apache-archiva.rb:28]: This should be easier
```suggestion
    ln_s wrapper/"bin/wrapper", libexec/"bin/wrapper"
    libext = OS.mac? ? "jnilib" : "so"
    ln_s wrapper/"lib/libwrapper.#{libext}", libexec/"lib/libwrapper.#{libext}"
    ln_s wrapper/"lib/wrapper.jar", libexec/"lib/wrapper.jar"
```

Comment (carlocab) [inline Formula/apache-archiva.rb:28]:
```suggestion
    ln_s (wrapper/"bin/wrapper").relative_path_from(libexec/"bin"), libexec/"bin"
    libext = OS.mac? ? "jnilib" : "so"
    ln_s (wrapper/"lib/libwrapper.#{libext}").relative_path_from(libexec/"lib"), libexec/"lib"
    ln_s (wrapper/"lib/wrapper.jar").relative_path_from(libexec/"lib"), libexec/"lib"
```
We want to make sure we create relative symlinks so this is reloctable. (`install_symlink` is not good for symlinks to other kegs.)

Though, maybe `brew` does this automatically when bottling; I don't remember. I can have a look at the bottle when it gets built.

File: Formula/apache-archiva.rb [hunk 5]
```diff
@@ -4,17 +4,28 @@ class ApacheArchiva < Formula
...
 
+  depends_on "ant" => :build
+  depends_on "java-service-wrapper"
   depends_on "openjdk"
 
   def install
     libexec.install Dir["*"]
+    rm_f Dir["#{libexec}/bin/wrapper*"]
+    rm_f Dir["#{libexec}/lib/libwrapper*"]
     (bin/"archiva").write_env_script libexec/"bin/archiva", JAVA_HOME: Formula["openjdk"].opt_prefix
```
Comment (carlocab) [inline Formula/apache-archiva.rb:22]:
```suggestion
    rm_f libexec.glob("bin/wrapper*")
    rm_f libexec.glob("lib/libwrapper*")
    (bin/"archiva").write_env_script libexec/"bin/archiva", Language::Java.java_home_env
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**